### ISSUE TITLE
How can I get separate weights and get custom gradients?

### ISSUE THREAD

**Issue by jnhelen** [keras-team/keras#11291]
I am implementing binary network in Keras so I need to compute the gradients on the binarized weights and then update the gradients on the weight before binarization. So the problem is how can I separate weights and apply different operations on them?  
![image](https://user-images.githubusercontent.com/33049481/46427436-06432e80-c774-11e8-8123-24c08d854900.png)
To convey my needs clearly, the back propagation part in the pic is what I want to do:)
( In tensorflow I could do that easily with add_collection and apply_gradients but I really hope to implement in Keras! Thanks!!!)
